File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

    let obj = {}
    
    for (let i = 0; i < nums.length; i++) {
        let curr = nums[i]
        let diff = target - curr
    
        if (obj[diff] >= 0) {           // make sure the key exists
            return [obj[diff], i]
        }
        obj[curr] = i

    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        length = len(nums)
        
        
        myMap = { nums[i]: i for i in range(length)}
        
        for i in range(length):  
            if ((target - nums[i]) not in myMap or myMap[target - nums[i]] == i):
                continue
            
            print(myMap)
            return [ i , myMap[target - nums[i]] ]
        